Assets/LeapMotion/Widgets/Scripts/Physics/LeapPhysicsBase.cs

```diff
@@ -0,0 +1,107 @@
+﻿using UnityEngine;
+
+namespace LMWidgets
+{
+  public enum LeapPhysicsState
+  {
+    Interacting, // Responsible for moving the widgets with the fingers
+    Reflecting // Responsible for reflecting widget information and simulating the physics
+  }
+
+  /// <summary>
+  /// Base class for physics. 
+  /// Handles state changes between Interacting and Reflecting.
+  /// </summary>
+  public abstract class LeapPhysicsBase : MonoBehaviour
+  {
+    protected LeapPhysicsState m_state = LeapPhysicsState.Reflecting;
+    protected GameObject m_target = null;
+    protected Vector3 m_pivot = Vector3.zero;
+    protected Vector3 m_targetPivot = Vector3.zero;
+
+    // Apply the physics interactions when the hand is no longer interacting with the object
+    protected abstract void ApplyPhysics();
+    // Apply interactions with the objects
+    protected abstract void ApplyInteractions();
+    // Apply constraints for the object (e.g. Constrain movements along a specific axis)
+    protected abstract void ApplyConstraints();
+
+    /// <summary>
+    /// Resets the pivots
+    /// </summary>
+    protected void ResetPivots()
```

Comment:
I'd be tempted to make this abstract (it would have to be if we made the pivots generic.